cli/oyster-cvm/src/commands/simulate.rs

```diff
@@ -0,0 +1,562 @@
+            hasher.update(enclave_path.as_os_str().as_encoded_bytes());
+            hasher.update(contents.len().to_le_bytes());
+            hasher.update(contents);
+
+            Ok(Some(hasher.finalize()))
+        })
+        .collect::<Result<Vec<_>>>()
+        .context("Failed to compute individual digest")?
+        .into_iter()
+        .flatten()
+        // accumulate further into a single hash
+        .fold(Sha256::new(), |mut hasher, param_hash| {
+            hasher.update(param_hash);
+            hasher
+        })
+        .finalize();
+
+    // Create and mount the init params digest into the container
+    let digest_file_path = "init_param_digest";
+    let mut file =
+        File::create(&digest_file_path).context("Failed to create temp init param digest file")?;
+    file.write_all(&digest)
+        .context("Failed to write to temp file")?;
+
+    let init_param_digest_host_path =
+        fs::canonicalize(&digest_file_path).context("Invalid init param digest path")?;
+    mount_args.append(&mut vec![
+        "-v".to_string(),
+        format!(
+            "{}:/app/init-params-digest",
+            init_param_digest_host_path.display()
+        ),
+    ]);
+
+    // Define memory configuration for the container based on user input
+    let mut config_args = vec![];
+    if args.container_memory.is_some() {
+        config_args.push(format!("--memory={}", args.container_memory.unwrap()));
+    } else {
+        match fetch_instance_memory(args.instance_type, args.arch).await {
+            Ok(memory) => config_args.push(format!("--memory={}MB", memory)),
+            Err(err) => info!("Failed to fetch instance memory: {}", err),
+        }
+    }
+
+    info!("Starting the dev container with user specified parameters");
+    let mut run_container = Command::new("docker")
+        .args(["run", "--privileged", "--rm", "-it"])
+        .args(&port_args)
+        .args(&mount_args)
+        .args(&config_args)
+        .args(["--name", &args.container_name])
+        .arg(&full_image_name)
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .spawn()
+        .context("Failed to start container")?;
+
+    let monitor_stats_task = thread::spawn(move || monitor_container_stats(&args.container_name));
+
+    let exit_status = run_container
+        .wait()
+        .context("Failed to wait on container")?;
+    info!("Dev container exited with status: {}", exit_status);
+
+    let _ = monitor_stats_task.join().unwrap();
+
+    if args.cleanup {
+        info!("Removing dev image...");
+        let remove_status = Command::new("docker")
+            .args(["rmi", &full_image_name])
+            .status()
+            .context("Failed to remove the pulled dev image")?;
+        info!("Dev image removed with status: {}", remove_status);
+    }
+
+    // Clean up the temporary files and directories created for simulation
+    if let Err(err) = fs::remove_dir_all(docker_images_temp_dir) {
```

Comment:
We can probably not delete this and use this as cache for future runs since docker save commands take the most time.